buildscripts/Dockerfile

```diff
@@ -0,0 +1,40 @@
+
+FROM ubuntu:14.04
+
+MAINTAINER Minio Community
+
+ENV GOLANG_TARBALL go1.4.2.linux-amd64.tar.gz
+
+ENV GOROOT /usr/local/go/
+ENV GOPATH /go-workspace 
+ENV PATH ${GOROOT}/bin:${GOPATH}/bin/:$PATH
+
+RUN apt-get update -y && apt-get install -y -q \
+		curl \
+		build-essential \
+		ca-certificates \
+		git yasm \
+		python-pip
+
+RUN curl -O -s https://storage.googleapis.com/golang/${GOLANG_TARBALL} && \
+		tar -xzf ${GOLANG_TARBALL} -C ${GOROOT%*go*} && \
+		rm ${GOLANG_TARBALL} 
+
+RUN pip install mkdocs
+
+RUN mkdir -p ${GOPATH}/src/github.com/minio-io/
```

Comment:
Instead of mkdir and git clone, it may be better to use ADD.
e.g.

ADD . ${GOPATH}/src/github.com/minio-io/minio

This does two things:

1. This will copy the entire local minio dir to that location and compile what is on your local system.
2. Eliminates caching issues. At present, docker will cache ENV and RUN commands listed above instead of running with the latest version.